Sources/WebSocket.swift

```diff
@@ -710,11 +710,11 @@ open class WebSocket : NSObject, StreamDelegate, WebSocketClient, WSStreamDelega
      Disconnect the stream object and notifies the delegate.
      */
     private func disconnectStream(_ error: Error?, runDelegate: Bool = true) {
-        if error == nil {
-            writeQueue.waitUntilAllOperationsAreFinished()
-        } else {
+        if error != nil {
             writeQueue.cancelAllOperations()
         }
+
+        writeQueue.waitUntilAllOperationsAreFinished()
```

Comment:
We should always wait, as cancel does not stop currently executing operations, or those queued for execution.

Reply:
>While the current thread is blocked, the receiver continues to launch already queued operations and monitor those that are executing. During this time, the current thread cannot add operations to the queue, but other threads may. Once all of the pending operations are finished, this method returns.

It seems like it is possible for that call to block the calling thread indefinitely if the socket is not already in a failed/error state.  Another thread communicating over the socket would still allowed, and `ping` or other custom protocol features would still occur. Might need additional protection of the socket to prevent further use...

Reply:
When there is no error, the behavior that this PR is proposing is the same exact behavior as previously. 

This PR essentially is only changing the behavior when there IS an error, per the diff.